src/query/ts/m3db/encoded_block_iterator.go

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3db
+
+import (
+	"time"
+
+	"github.com/m3db/m3/src/dbnode/encoding"
+	"github.com/m3db/m3/src/query/block"
+	"github.com/m3db/m3/src/query/functions/utils"
+	"github.com/m3db/m3/src/query/models"
+	"github.com/m3db/m3/src/query/storage"
+)
+
+type consolidationSettings struct {
+	consolidationFn ConsolidationFunc
+	currentTime     time.Time
+	bounds          *models.Bounds
+}
+
+type encodedBlock struct {
+	lastBlock            bool
```

Comment:
this is weird

Reply:
This will be necessary for when series are split into blocks; the last block has a distinct that it excludes any values at EndTime, whereas other blocks include it.

Reply:
i'm not sure I follow. Shouldn't all blocks exclude the EndTime since endTime for block1 is same as start time for block2 so if you include endtime for block1, its double counted ?

Reply:
Because of the way the internal blockIterators that make up a seriesIterator are structured, values at the block boundary can go in either block; there will only be 1 value but it may be in either one. Because of this, we always want to take values at the end in all blocks except the last, which would be the exclusive end of the entire query.

Reply:
if you count that point within block1, what would be the timestamp associated with it ? If its timestamp is the end, then it might get excluded by downstream since they exclude bounds.End

Reply:
Downstream consumers are going to have to be changed when the multiblock read is enabled in either case, which is when this will be necessary

Reply:
Can you add some comments here ? Anyone else looking at the code may have similar questions.

Reply:
Added a brief comment here about why this bool exists; will expand on why when the logic for choosing to add/drop last point is added